apps/modernization-ui/src/permissions/useSystemManagementAccess.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo } from 'react';
+
+const SYSTEM_MANAGEMENT_PERMISSIONS = [
+    'EPILINKADMIN-SYSTEM',
+    'VIEWELRACTIVITY-OBSERVATIONLABREPORT',
+    'SRTADMIN-SYSTEM',
+    'VIEWPHCRACTIVITY-CASEREPORTING',
+    'IMPORTEXPORTADMIN-SYSTEM',
+    'REPORTADMIN-SYSTEM',
+    'ALERTADMIN-SYSTEM'
+];
+
+const useSystemManagementAccess = (permissions: string[] | undefined | null) => {
+    const hasAccess = useMemo(() => {
+        return permissions && permissions.some((permission) => SYSTEM_MANAGEMENT_PERMISSIONS.includes(permission));
```

Comment:
The `some` and the `includes` causes multiple iterations over the `SYSTEM_MANAGEMENT_PERMISSIONS` array.  The `reduce` used by `usePatientProfilePermissions` is done to prevent that.

Reply:
will use it